src/Data/Nat/Combinatorics.agda

```diff
@@ -85,3 +96,115 @@ nCn≡1 n = begin-equality
   1              ∎
   where instance
     _ = n !≢0
+
+nC1≡n : ∀ n → n C 1 ≡ n
+nC1≡n zero    = refl
+nC1≡n n@(suc n-1) = begin-equality
+  n C 1          ≡⟨ nCk≡nPk/k!  (s≤s (z≤n {n-1})) ⟩
+  (n P 1) / 1 !  ≡⟨ n/1≡n (n P 1) ⟩
+  n P 1          ≡⟨ nP1≡n n ⟩
+  n              ∎
+
+
+------------------------------------------------------------------------
+-- Arithmetic of (n C k)
+
+nCk+nC[k+1]≡[n+1]C[k+1] : ∀ n k → n C k + n C (suc k) ≡ suc n C suc k
+nCk+nC[k+1]≡[n+1]C[k+1] n k with <-cmp k n
+{- case k>n, in which case 1+k>1+n>n -}
+... | tri> _ _ k>n = begin-equality
+  n C k + n C (suc k) ≡⟨ cong (_+ (n C (suc k))) (k>n⇒nCk≡0 k>n) ⟩
+  0 + n C (suc k)     ≡⟨⟩
+  n C (suc k)         ≡⟨ k>n⇒nCk≡0 (m<n⇒m<1+n k>n) ⟩
+  0                   ≡˘⟨ k>n⇒nCk≡0 (s<s k>n) ⟩
+  suc n C suc k       ∎
+{- case k≡n, in which case 1+k≡1+n>n -}
+... | tri≈ _ k≡n _ rewrite k≡n = begin-equality
+  n C n + n C (suc n) ≡⟨ cong (n C n +_) (k>n⇒nCk≡0 (n<1+n n)) ⟩
+  n C n + 0           ≡⟨ +-identityʳ (n C n) ⟩
+  n C n               ≡⟨ nCn≡1 n ⟩
+  1                   ≡˘⟨ nCn≡1 (suc n) ⟩
+  suc n C suc n       ∎
+{- case k<n, in which case 1+k<1+n and there's arithmetic to perform -}
+... | tri< k<n _ _ = begin-equality
+  n C k + n C (suc k)
+    ≡⟨ cong (n C k +_) (nCk≡n!/k![n-k]! k<n)  ⟩
+  n C k + (n! / d[k+1])
+    ≡˘⟨ cong (n C k +_) (m*n/m*o≡n/o (n ∸ k) n! d[k+1]) ⟩
+  n C k + [n-k]*n!/[n-k]*d[k+1]
+    ≡⟨ cong (_+ [n-k]*n!/[n-k]*d[k+1]) (nCk≡n!/k![n-k]! k≤n) ⟩
+  n! / d[k] + _
+    ≡˘⟨ cong (_+ [n-k]*n!/[n-k]*d[k+1]) (m*n/m*o≡n/o (suc k) n! d[k]) ⟩
+  (suc k * n!) / [k+1]*d[k] + _
+    ≡⟨ cong (((suc k * n!) / [k+1]*d[k]) +_) (/-congʳ [n-k]*d[k+1]≡[k+1]*d[k]) ⟩
+  (suc k * n!) / [k+1]*d[k] + ((n ∸ k) * n! / [k+1]*d[k])
+    ≡˘⟨ +-distrib-/-∣ˡ _ (*-monoʳ-∣ (suc k) (k![n∸k]!∣n! k≤n)) ⟩
+  ((suc k) * n! + (n ∸ k) * n!) / [k+1]*d[k]
+    ≡˘⟨ cong (_/ [k+1]*d[k]) (*-distribʳ-+ (n !) (suc k) (n ∸ k)) ⟩
+  ((suc k + (n ∸ k)) * n !) / [k+1]*d[k]
+    ≡⟨ cong (λ z → z * n ! / [k+1]*d[k]) [k+1]+[n-k]≡[n+1] ⟩
+  ((suc n) * n !) / [k+1]*d[k]
+    ≡˘⟨ /-congʳ (*-assoc (suc k) (k !) ((n ∸ k) !)) ⟩
+  ((suc n) * n !) / (((suc k) * k !) * (n ∸ k) !)
+    ≡⟨⟩
+  suc n ! / (suc k ! * (suc n ∸ suc k) !)
+    ≡˘⟨ nCk≡n!/k![n-k]! [k+1]≤[n+1] ⟩
+  suc n C suc k ∎
+  where
+    k≤n                     : k ≤ n
+    k≤n                     = <⇒≤ k<n
+
+    [k+1]≤[n+1]             : suc k ≤ suc n
+    [k+1]≤[n+1]             = s≤s k≤n
+
+    [k+1]+[n-k]≡[n+1]       : (suc k) + (n ∸ k) ≡ suc n
+    [k+1]+[n-k]≡[n+1]       = m+[n∸m]≡n {suc k} [k+1]≤[n+1]
+
+    [n-k]                   = n ∸ k
+    [n-k-1]                 = n ∸ suc k
+
+    [n-k-1]+1≡[n-k]         : suc [n-k-1] ≡ [n-k]
+    [n-k-1]+1≡[n-k]         = [m-n-1]+1≡[m-n] k<n
+
+    n!                      = n !
+    k!                      = k !
+    [k+1]!                  = (suc k) !
+    [n-k]!                  = [n-k] !
+    [n-k-1]!                = [n-k-1] !
+
+    [n-k]*[n-k-1]!≡[n-k]!   : [n-k] * [n-k-1]! ≡ [n-k]!
```

Comment:
I feel like the two non-trivial sub-lemmas should be lifted to the top level?

Reply:
OK, that's probably fair. Any subsequent use of this module might indeed find such things useful!

Reply:
I've tried this; one argument against doing so is that too much let-bound state is shared between the lemmas and their use in the third branch of the main result. But I can see the argument for their being independenlty interesting.